modules/liburing/2.5/patches/add_build_file.patch

```diff
@@ -0,0 +1,63 @@
+--- /dev/null
++++ a/BUILD.bazel
+@@ -0,0 +1,60 @@
++genrule(
++    name = "generate_headers",
++    srcs = [
++        "Makefile.common",
++        "liburing.spec",
++    ],
++    outs = [
++        "config-host.h",
++        "src/include/liburing/compat.h",
++        "src/include/liburing/io_uring_version.h",
++    ],
++    cmd = """
++            export CC=$(CC) CXX=$(CC)++
++            cd $$(dirname $(location configure))
++            mkdir -p src/include/liburing
++            ./configure
```

Comment:
Is this needed? It presumably performs non-hermetic feature detection of the machine that this action is executed on, which is not necessarily the machine that the build targets.

Reply:
I am not sure how to overcome this. I tried with statically adding the generated files, but the presubmit failures came as expected due to different host kernel versions.

Different kernel version come with different kernel headers (which remain a host dependency) and the configure step here creates a compatibility layer. While the result is portable, the intermediate is host specific. I added a comment to explain the situation.

Do you want me marking those rules `tags=["local"]` ?

Reply:
If the result is portable then the way it is set up looks right and we don't need any tags. Thanks for adding the comment!